vms/platformvm/service.go

```diff
@@ -1233,13 +1233,10 @@ func (s *Service) AddValidator(req *http.Request, args *AddValidatorArgs, reply
 	}
 
 	s.vm.ctx.Lock.Lock()
-	defer s.vm.ctx.Lock.Unlock()
-
 	user, err := keystore.NewUserFromKeystore(s.vm.ctx.Keystore, args.Username, args.Password)
 	if err != nil {
 		return err
 	}
-	defer user.Close()
```

Comment:
We defer the closing of the DB here to prevent it from leaking if an error occurs mid-way through the function. ex: user has no keys or insufficient funds.

(Same comment applies below)